core/src/test/scala/matryoshka/runners/package.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2014–2016 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package matryoshka
+
+import Recursive.ops._
+
+import scala.Unit
+import scala.Predef.implicitly
+
+import org.specs2.execute._
+import org.specs2.matcher._
+import org.specs2.scalaz._, ScalazMatchers._
+import scalaz._, Scalaz._
+
+package object runners extends Spec {
+  abstract class RecRunner[F[_], A] {
+    // NB: This is defined as a function to make the many definition sites
+    //     slightly shorter.
+    def run[T[_[_]]: Recursive]: T[F] => MatchResult[A]
+  }
+  def testRec[F[_], A](t: Fix[F], r: RecRunner[F, A])(implicit F: Functor[F]):
+      MatchResult[A] = {
+    r.run[Fix].apply(t) and
+    r.run[Mu].apply(t.convertTo[Mu]) and
+    r.run[Nu].apply(t.convertTo[Nu]) and
+    r.run[Cofree[?[_], Unit]].apply(t.convertTo[Cofree[?[_], Unit]](F, cofreeCorecursive[Unit]))
+  }
+
+  abstract class CorecRunner[M[_], F[_], A] {
+    def run[T[_[_]]: Corecursive](implicit Eq: Equal[T[F]], S: Show[T[F]]):
+        A => MatchResult[M[T[F]]]
+  }
+  def testCorec[M[_], F[_]: Functor, A](
+    a: A, r: CorecRunner[M, F, A])(
+    implicit Eq0: Equal ~> λ[α => Equal[F[α]]], S0: Show ~> λ[α => Show[F[α]]]):
+      Result =
+    r.run[Fix].apply(a).toResult and
+    r.run[Mu].apply(a).toResult and
+    r.run[Nu].apply(a).toResult and
+    // NB: No Equal for Free, so we can’t test it.
+    // r.run[Free[?[_], Unit]](freeCorecursive[Unit], implicitly, implicitly).apply(a).toResult and
```

Comment:
I'd say there's no point in showing the implementation if it isn't possible to make it work, especially since it's easy to guess what it would look like from the next line.